src/preamble.js

```diff
@@ -2034,6 +2034,10 @@ function integrateWasmJS() {
     wasmTextFile = Module['locateFile'](wasmTextFile);
     wasmBinaryFile = Module['locateFile'](wasmBinaryFile);
     asmjsCodeFile = Module['locateFile'](asmjsCodeFile);
+  } else {
+    wasmTextFile = Module['scriptDirectory'] + wasmTextFile;
+    wasmBinaryFile = Module['scriptDirectory'] + wasmBinaryFile;
+    asmjsCodeFile = Module['scriptDirectory'] + asmjsCodeFile;
```

Comment:
Here `scriptDirectory` is applied as an alternative to `locateFile`. I think a good semantic would be that `scriptDirectory` would be applied on top of `locateFile` or the other prefixes. That way if one prepends a relative URL from a `locateFile` function, e.g. `Module.locateFile = function(url) { return 'Release/' + url' }`, then `scriptDirectory` will still be applied to it, producing the CWD routing result.

Reply:
This is exactly the same semantic that was used in other places before with options like `memoryInitializerPrefixURL`.

Chaining these options makes things more complex and increases cognitive load when you're trying to figure out what the final path would be. For such a niche use case I don't think over engineering makes a lot of sense.

Reply:
> Chaining these options makes things more complex and increases cognitive load

Returning a relative path from `Module.locateFile()` is a use case that has worked before, and with code like above, if one defined `Module.locateFile()`, it would then not receive the CWD fix for Node.js. Treating relative return values from `Module.locateFile()` with respect to `Module['scriptDirectory']` will allow those use cases to also work.

Reply:
> if one defined Module.locateFile(), it would then not receive the CWD fix for Node.js

Because it is not supposed to! I don't really see why you'd want some additional tweaks to be added to the result of location function that you've specified. It doesn't make any sense to me.